packages/hydrogen/src/customer/customer.ts

```diff
@@ -201,11 +204,22 @@ export function createCustomerClient({
       throwGraphQLError({...errorOptions, errors});
     }
 
+    let bodyJSON;
     try {
-      return parseJSON(body);
+      bodyJSON = parseJSON(body);
     } catch (e) {
       throwGraphQLError({...errorOptions, errors: [{message: body}]});
     }
+
+    if (bodyJSON.errors?.length) {
```

Comment:
Def a consideration. I made this PR to be more inline with Storefront API client but kept it in Draft because I am still debating on how to move forward.

Since the Customer Account API is still in preview and client is not being use widely. I am considering converting the return value & error handling to look more similar to [apollo.query](https://www.apollographql.com/docs/react/api/core/ApolloClient/#query) so it would be more familiar for developrs.

Reply:
Let's discuss this a bit more so we can have a general approach to handling graphql errors. I have a similar PR that captures half errors on storefront client that I didn't merge yet https://github.com/Shopify/hydrogen/pull/1477